packages/office-ui-fabric-react/src/components/TeachingBubble/examples/TeachingBubble.Basic.Example.tsx

```diff
@@ -1,74 +1,42 @@
 import * as React from 'react';
-
 import { DefaultButton, IButtonProps } from 'office-ui-fabric-react/lib/Button';
 import { TeachingBubble } from 'office-ui-fabric-react/lib/TeachingBubble';
+import { useBoolean } from '@uifabric/react-hooks';
 
-export interface ITeachingBubbleBasicExampleState {
-  isTeachingBubbleVisible?: boolean;
-}
-
-export class TeachingBubbleBasicExample extends React.Component<{}, ITeachingBubbleBasicExampleState> {
-  private _menuButtonElement: HTMLElement;
-
-  constructor(props: {}) {
-    super(props);
-
-    this._onDismiss = this._onDismiss.bind(this);
-    this._onShow = this._onShow.bind(this);
+const examplePrimaryButtonProps: IButtonProps = {
+  children: 'Try it out',
+};
 
-    this.state = {
-      isTeachingBubbleVisible: false,
-    };
-  }
-
-  public render(): JSX.Element {
-    const { isTeachingBubbleVisible } = this.state;
-    const examplePrimaryButton: IButtonProps = {
-      children: 'Try it out',
-    };
-    const exampleSecondaryButtonProps: IButtonProps = {
+export const TeachingBubbleBasicExample: React.FunctionComponent = () => {
+  const [teachingBubbleVisible, { toggle: toggleTeachingBubbleVisible }] = useBoolean(false);
+  const exampleSecondaryButtonProps: IButtonProps = React.useMemo(
```

Comment:
useMemo should be used for expensive functions. This is a very cheap calculation and using useMemo would probably be confusing for new developer looking at the example code.

Reply:
nice work getting useMemo working though...i had to look it up to remember how it worked!

Reply:
Caleb actually added this in response to my earlier comment--we discussed this awhile back (unfortunately don't remember who/where/when...) and decided that even though useMemo is not strictly necessary here, we want to demonstrate it (and similar patterns) in our examples to hopefully encourage their use when it actually matters.

Reply:
Also, useMemo is important here to reduce the frequency that the button props object mutates. For that reason it should definitely stay in this case.

Reply:
We'll stick with them for now.